src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2127,6 +2128,21 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                                                  INS_SCALABLE_OPTS_LSL_N);
                 break;
             }
+
+            case NI_Sve_InsertIntoShiftedVector:
+            {
+                assert(isRMW);
+                if (targetReg != op1Reg)
+                {
+                    assert(targetReg != op2Reg);
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg,
```

Comment:
Does this work with both `op1Reg` being gpr or SIMD/FP?

Reply:
I think so, though I'm struggling to get the JIT to use a gpr for `op1Reg` under the stress modes -- I'm only ever getting the "easy" case, where when `op1Reg` and `targetReg` differ, `op1Reg` is already a vector register, so we're moving from a vector reg to a vector reg. Since the first argument in `Sve.InsertIntoShiftedVector<T>` is of type `Vector<T>`, we'd expect `op1Reg` to always be a vector register, right? I could add an assert here clarify this.

Reply:
Actually I meant that for `op2Reg`. Sorry. So ideally, where you have `double` or `float` as 2nd argument, we should have `op2Reg` as SIMD/floating point and otherwise should be gpr. Can you verify that please? For `op1Reg` it should always be scalable register and we already assert that in emitter.

Reply:
No worries, I've added an assert to check this. Stress tests for unoptimized and optimized tests are still passing.

Reply:
Do you mind sharing small section of disassembly for both the categories?

Reply:
Not at all. Here's a snippet from the `double` tests:
```
            ldr     q16, [x0]
            ldr     d17, [fp, #0x30]    // [V01 loc0]
            insr    z16.d, d17
```

And from the `uint` tests:
```
            ldr     q16, [x0]
            ldr     w0, [fp, #0x34]     // [V01 loc0]
            insr    z16.s, w0
```